web/src/main/java/org/springframework/security/web/util/UrlUtils.java

```diff
@@ -61,7 +65,7 @@ public static String buildFullRequestUrl(String scheme, String serverName,
 		}
 		else if ("https".equals(scheme)) {
 			if (serverPort != 443) {
-				url.append(":").append(serverPort);
```

Comment:
See `java.lang.AbstractStringBuilder#append(java.lang.String)` vs `java.lang.AbstractStringBuilder#append(char)`.
The latter just does `value[count++] = c;` while the String version calls `java.lang.String#getChars(int, int, char[], int)` which always does `System.arraycopy(value, srcBegin, dst, dstBegin, srcEnd - srcBegin);`.
So the `char` version is much cheaper. But the gain is noticeable only when there are many concatenations.